### PR TITLE
Add multisite support for setup tasks

### PR DESCRIPTION
**Pull request by bobbygryzynger**
Addresses aspects of #799.

Changes proposed:
- Allow multisite configurations to be supplied in per-site yaml files.
- Perform setup tasks for each multisite.

### CODE REVIEW

**Comment by bobbygryzynger** [comment]
@grasmash - I've taken a stab here at breaking up #827 a bit, let me know what you think.

File: phing/tasks/properties.xml [hunk 1]
```diff
@@ -56,7 +56,43 @@
...
-  <echo level="verbose">Executing commands against multisite "${multisite.name}"</echo>
+  <!-- Set the default site to operate against. -->
+  <property name="site.name" value="default" override="false"/>
+
+  <!-- Load multisite-specific overrides. -->
+  <if>
+    <available file="${blt.config-dirs.sites}"/>
+    <then>
+      <!-- Build the site list from *.yml files in the blt/sites directory. -->
+      <exec dir="${blt.config-dirs.sites}" command="ls *.yml | tr -d '\n' | sed s/\.yml/,/g" outputProperty="multisite.names"/>
```
Comment (grasmash) [inline phing/tasks/properties.xml]: I think we may need a more blt-specific pattern, since this could pick up unrelated yml files. Perhaps project.site.yml.

**Comment by grasmash** [comment]
Thanks for breaking this into something smaller. I think that the last step before merging is to ensure that there's some type of test coverage. Perhaps we can add a default project.site.yml to a directory during BLT's .travis.yml execution and assert that it has been parsed and generated the expected settings files.

**Comment by kylebrowning** [comment]
👍

**Comment by bobbygryzynger** [comment]
@grasmash - working on adding some tests here, but ran into #1011.

**Comment by bobbygryzynger** [comment]
@grasmash - I've put some additional testing in place here. I wasn't entirely sure how to go about testing that site properties are being correctly parsed, but take a look at `PropertiesTest.php` and see if you feel like it's a reasonable approach.
